text/0100-sensitive-data-handling.md

```diff
@@ -0,0 +1,103 @@
+less-sophisticated regex that bluntly removes too much from the query), but in other cases we may need to, e.g., throw an internal exception and 
+interpret that as a requirement to not add the 
+attribute in the first place.  In pseudo-code:
+```
+  try {
+      span.setAttribute("sensitive.field", normalize(sensitiveValue));
+  } catch (CannotNormalizeException e) {
+     // don't emit sensitive.field since we can't normalize it safely
+  }
+```
+
+This work will affect all auto-instrumenting libraries, as well as any official (or recommended) manually wrapped libraries that are 
+produced as part of the ecosystem.  As appropriate to each such library, features designed to properly handle sensitive data should
+be highlighted in the documentation.  Sticking with the SQL example, for the Java auto library this could take the form of an additional column
+on the "Supported Frameworks" table with a hyperlink to the documentation describing exactly what we do:
+
+| Library/Framework                                                                                    | Versions                       | Notes                                      |
+|------------------------------------------------------------------------------------------------------|--------------------------------|--------------------------------------------|
+| [JDBC](https://docs.oracle.com/en/java/javase/11/docs/api/java.sql/java/sql/package-summary.html)    | Java 7+                        | (link to docs about sql normalization)     |
+
+
+## Trade-offs and mitigations
+
+In some cases, proper "safe by default" behavior might involve more intensive processing than can reasonably be performed inline with the application.  There have 
+already been some discussions about having span (and attribute) processing logic run in-process but off-thread in the application (i.e., by a configurable span 
+processing pipeline, which would use separate threads for its work).  For sql normalization (my first/example concern), I don't believe this is necessary, but 
+further work on this issue may put pressure here.
+
+One drawback of this "safe by default" approach is that it puts a configuration burden on some users who are not handling sensitive data but
+want to have as much visibility as possible into their system.  Staying with the sql normalization example however, the vast majority of sql in the world doesn't 
+have different performance characteristics based on the exact values passed in, but on the number of them, or the complexity of the `WHERE` clauses, 
+the number of values in an `IN` list, presence of an index on fields, etc.
+
+Some might consider doing most of this work at the collector, rather than in-process with the app.  For users who are already exposed to this 
+issue, I don't think the statement "yes, by default we scrub the credit card number immediately after it is sent over the wire to our collector" is a very reassuring one.
```

Comment:
I'm skeptical about the performance impact on the host application, somewhat, but I'm even less convinced that we're going to build out scrubbers for PII in every client language. 

This seems to suggest a configuration where an OTel collector is co-located in the side-car configuration, so that it resides in the same security domain. Then the statement is "yes, by default we scrub the credit card number immediately after it crosses outside your process a locally running sidecar."

Reply:
+1. Making the promise that scrubbing will be done in-process in every language is a very large scope creep. Doing it once in a sidecar or central tier is much more sustainable and manageable.